cluster/kubernetes/kubernetes.go

```diff
@@ -42,19 +43,16 @@ type extendedClient struct {
 
 type apiObject struct {
 	bytes    []byte
-	Version  string `yaml:"apiVersion"`
 	Kind     string `yaml:"kind"`
 	Metadata struct {
 		Name      string `yaml:"name"`
 		Namespace string `yaml:"namespace"`
 	} `yaml:"metadata"`
 }
 
-func (obj *apiObject) namespaceOrDefault() string {
-	if obj.Metadata.Namespace == "" {
-		return "default"
-	}
-	return obj.Metadata.Namespace
+func (o *apiObject) hasDefaultNamespace() bool {
```

Comment:
I thought the problem was with manifests that did not give a namespace. If "default" is supplied, that should be fine, as with any other namespace. In other words, what you _really_ care about here is whether it's supplied or not.

Reply:
Correctamundo. Addressed in an unpushed commit.